### PR TITLE
test(react): adding cypress visual regression tests for IcSearchBar

### PR DESCRIPTION
**Pull request by MI6-286**
Adding cypress visual regression tests for IcSearchBar 

## Summary of the changes
Migrated loki screenshots for IcSearchBar to Cypress visual regression tests.

## Related issue
#1891

### CODE REVIEW

File: packages/react/src/component-tests/IcSearchBar/IcSearchBar.cy.tsx [hunk 1]
```diff
@@ -65,65 +77,355 @@ describe("IcSearchBar", () => {
...
+  it("should render search bar within a form", () => {
+    mount(<InAForm />);
+
+    cy.checkHydrated(SEARCH_SELECTOR);
+
+    cy.checkA11yWithWait();
+    cy.compareSnapshot({
+      name: "in-a-form",
+      testThreshold: setThresholdBasedOnEnv(DEFAULT_TEST_THRESHOLD + 0.021),
+    });
```
Comment (gd2910) [inline packages/react/src/component-tests/IcSearchBar/IcSearchBar.cy.tsx:257]: Not sure we need a snapshot for this state?

File: packages/react/src/component-tests/IcSearchBar/IcSearchBar.cy.tsx [hunk 2]
```diff
@@ -65,65 +77,355 @@ describe("IcSearchBar", () => {
...
+  it("should render search bar with query search mode", () => {
+    mount(<QuerySearchMode />);
+
+    cy.checkHydrated(SEARCH_SELECTOR);
+
+    //cy.checkA11yWithWait();
+    cy.compareSnapshot({
+      name: "query-search-mode",
+      testThreshold: setThresholdBasedOnEnv(DEFAULT_TEST_THRESHOLD + 0.021),
+    });
```
Comment (gd2910) [inline packages/react/src/component-tests/IcSearchBar/IcSearchBar.cy.tsx:359]: Another one I'm not sure on the benefit of a snapshot in this state?

File: packages/react/src/component-tests/IcSearchBar/IcSearchBar.cy.tsx [hunk 3]
```diff
@@ -11,23 +11,38 @@ import {
...
 import "cypress-axe";
 
 const SEARCH_SELECTOR = "ic-search-bar";
+const SEARCH_INPUT = 'input[type="text"]';
 const TEXT_FIELD_SELECTOR = "ic-text-field";
-const DEFAULT_TEST_THRESHOLD = 0.042;
+const DEFAULT_TEST_THRESHOLD = 0.02;
 
-describe("IcSearchBar", () => {
+describe("IcSearchBar end-to-end tests", () => {
```
Comment (GCHQ-Developer-299) [inline packages/react/src/component-tests/IcSearchBar/IcSearchBar.cy.tsx:40]: Sorry to be a pedant, but none of our other Cypress tests have 'end-to-end tests' in the description here

Comment (MI6-286) [reply packages/react/src/component-tests/IcSearchBar/IcSearchBar.cy.tsx:40]: Some of the newer ones do (IcTextField and IcTabs) and others have something like E2E. Where we're separating out the 'end-to-end' and visual regression tests I think it's good to have the longer description.

**Review by gd2910 — APPROVED** [review]

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Pull request merged**